### PR TITLE
FIX Allow a null object when getting changeset edit link

### PR DESCRIPTION
**Pull request by GuySartorelli**
`ChangeSetItem::getObjectInStage()` returns the value of `DataList::byID()` which is explicitly nullable. It needs to handle null values.

Otherwise we get this error after https://github.com/silverstripe/silverstripe-assets/pull/707 gets merged:

> ERROR [Emergency]: Uncaught Error: Call to a member function CMSEditLink() on null
> IN GET /admin/campaigns/sets
> Line 537 in /var/www/html/vendor/silverstripe/versioned/src/ChangeSetItem.php

## Issue

- https://github.com/silverstripe/silverstripe-campaign-admin/issues/393

### CODE REVIEW

File: src/ChangeSetItem.php [hunk 1]
```diff
@@ -170,7 +170,7 @@ public function getChangeType()
      * If the object isn't versioned it will return the normal record.
      *
      * @param string $stage
-     * @return DataObject|Versioned|RecursivePublishable Object in this stage (may not be Versioned)
+     * @return DataObject|Versioned|RecursivePublishable|null Object in this stage (may not be Versioned)
```
Comment (ChloeSartorelli) [inline src/ChangeSetItem.php:173]: This change just reflects reality - this method returns the value from `DataList::byID()` which is explicitly nullable.

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Add a test that `CMSEditLink()` returns `null` when the underlying object can’t be loaded

**Comment by GuySartorelli** [comment]
Test added

**Review by emteknetnz — APPROVED** [review]
